sharding-core/sharding-core-merge/src/main/java/org/apache/shardingsphere/core/merge/dal/show/ShowDatabasesMergedResult.java

```diff
@@ -30,22 +31,42 @@
 import java.sql.SQLXML;
 import java.util.Collections;
 import java.util.List;
+import org.apache.shardingsphere.core.rule.ShardingRule;
 
 /**
  * Merged result for show databases.
  *
  * @author chenqingyang
  */
-@RequiredArgsConstructor
 public final class ShowDatabasesMergedResult extends LocalMergedResultAdapter implements MergedResult {
     
     private final List<String> schemas;
     
     private int currentIndex;
-    
+
     public ShowDatabasesMergedResult() {
         this(Collections.singletonList(ShardingConstant.LOGIC_SCHEMA_NAME));
     }
+
+    //for proxy backend, just return logic database
+    public ShowDatabasesMergedResult(final List<String> schemas) {
+        this.schemas = schemas;
+        this.currentIndex = 0;
+    }
+
+    public ShowDatabasesMergedResult(final ShardingRule shardingRule, final List<QueryResult> queryResults)throws SQLException {
+        this.schemas = new ArrayList<>();
```

Comment:
If the List is used to for each only, we prefer using LinkedList, otherwise the ArrayList should have a certainly initial value to avoid recalculate the size and copy elements.
If using LinkedList, we prefer using Collection as variable type to avoid developer use `get(i)`, because random visit is too slow of LinkedList.

Reply:
this constructor not easy to arraylist, beacuse the QueryResult SPI not size() function. so i will refactor it..